File: src/main/scala/lint/jobs/Benchmark.scala


package lint.jobs

import org.apache.spark.{SparkContext,SparkConf}
import pprint.pprintln


object Benchmark {

  lazy val sc = new SparkContext(new SparkConf)

  def main(args: Array[String]) {

    val data = sc.parallelize(0 until 480)

    val res = data
      .map(i => {
        new Thread
        Thread.sleep(1000)
        i+1
      })
      .collect()

    pprintln(res)

  }

}

File: code/lint/jobs/benchmark.py


import time

from lint.utils import get_spark


def work(i):
    time.sleep(1)
    return i+1


def main():
    sc, _ = get_spark()
    data = sc.parallelize(range(100))
    result = data.map(work).collect()
    print(result)


if __name__ == '__main__':
    main()
